### ISSUE TITLE
Unable to install no_full_mz_array branch from Github

### ISSUE THREAD

**Issue by wkumler** [PNNL-m-q/mzapy#25]
Hi, thanks for your work on this package! I was directed here after being unable to open an .mza file after conversion from an Orbitrap .mzML due to the error of `KeyError: "Unable to synchronously open object (object 'Full_mz_array' doesn't exist)"` (https://github.com/PNNL-m-q/mza/issues/2). There, @aivett recommended installing the no_full_mz_array branch since they knew you'd been working on this issue. Unfortunately, none of the methods I've tried have allowed me to install the branch and instead throw the error of `Invalid version: '1.8.0.no_full_mz_array_2'`.

I tried installing via Git with `pip install git+https://github.com/PNNL-m-q/mzapy.git@no_full_mz_array#egg=mzapy`, via raw pip with: `pip install --upgrade https://github.com/PNNL-m-q/mzapy/tarball/no_full_mz_array`, and with a direct clone and then `pip install .` from within the package directory and all three methods reported the below error:

```
$ pip install .
Processing c:\users\willi\desktop\mzapy
  Installing build dependencies: started
  Installing build dependencies: finished with status 'done'
  Getting requirements to build wheel: started
  Getting requirements to build wheel: finished with status 'error'
  error: subprocess-exited-with-error

  Getting requirements to build wheel did not run successfully.
  exit code: 1

  [56 lines of output]
  running egg_info
  Traceback (most recent call last):
    File "C:\Users\willi\AppData\Local\Temp\pip-build-env-4payfuyf\overlay\Lib\site-packages\setuptools\_normalization.py", line 61, in safe_version
      return str(packaging.version.Version(v))
                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    File "C:\Users\willi\AppData\Local\Temp\pip-build-env-4payfuyf\overlay\Lib\site-packages\setuptools\_vendor\packaging\version.py", line 202, in __init__
      raise InvalidVersion(f"Invalid version: {version!r}")
  packaging.version.InvalidVersion: Invalid version: '1.8.0.no_full_mz_array_2'
```

Any recommendations for how to get this branch installed for use with Orbitrap data?

**Comment by dylanhross** [comment]
Hi There, 

Thanks for bringing this up. I had not tested pip installing with this modified branch, and it turns out pip expects package versions to be in a [specific format](https://peps.python.org/pep-0440/) to be pip installable. Version string updated to be compatible, tested in a venv (see below) and that seems to have fixed it. Let me know if you run into any more issues. 

-Dylan

```
(testenv) ╭─ross200@WE50940 ~/Desktop/mzapy_test 
╰─$ python3 -c "from mzapy import __version__; print(__version__)"
Traceback (most recent call last):
  File "<string>", line 1, in <module>
ModuleNotFoundError: No module named 'mzapy'

(testenv) ╭─ross200@WE50940 ~/Desktop/mzapy_test 
╰─$ pip install git+https://github.com/PNNL-m-q/mzapy.git@no_full_mz_array#egg=mzapy                                1 ↵
Collecting mzapy
  Cloning https://github.com/PNNL-m-q/mzapy.git (to revision no_full_mz_array) to /private/var/folders/wy/94m7dyf92f519_z6_pt64cg80000gn/T/pip-install-zwj2t2xs/mzapy_f87f96811b3c435386667fd1d1c960e7
  Running command git clone --filter=blob:none --quiet https://github.com/PNNL-m-q/mzapy.git /private/var/folders/wy/94m7dyf92f519_z6_pt64cg80000gn/T/pip-install-zwj2t2xs/mzapy_f87f96811b3c435386667fd1d1c960e7
  Running command git checkout -b no_full_mz_array --track origin/no_full_mz_array
  Switched to a new branch 'no_full_mz_array'
  branch 'no_full_mz_array' set up to track 'origin/no_full_mz_array'.
  Resolved https://github.com/PNNL-m-q/mzapy.git to commit 2dfb1e8bf9fd571015353b1a2a0130c4fd9de36f
  Installing build dependencies ... done
  Getting requirements to build wheel ... done
  Preparing metadata (pyproject.toml) ... done
Collecting h5py (from mzapy)
  Downloading h5py-3.12.1-cp312-cp312-macosx_11_0_arm64.whl.metadata (2.5 kB)
Collecting hdf5plugin (from mzapy)
  Using cached hdf5plugin-5.0.0-py3-none-macosx_10_13_universal2.whl.metadata (4.7 kB)
Collecting pandas (from mzapy)
  Downloading pandas-2.2.3-cp312-cp312-macosx_11_0_arm64.whl.metadata (89 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 89.9/89.9 kB 2.8 MB/s eta 0:00:00
Collecting numpy (from mzapy)
  Downloading numpy-2.1.3-cp312-cp312-macosx_14_0_arm64.whl.metadata (62 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 62.0/62.0 kB 7.2 MB/s eta 0:00:00
Collecting scipy (from mzapy)
  Using cached scipy-1.14.1-cp312-cp312-macosx_14_0_arm64.whl.metadata (60 kB)
Collecting matplotlib (from mzapy)
  Using cached matplotlib-3.9.2-cp312-cp312-macosx_11_0_arm64.whl.metadata (11 kB)
Collecting contourpy>=1.0.1 (from matplotlib->mzapy)
  Downloading contourpy-1.3.1-cp312-cp312-macosx_11_0_arm64.whl.metadata (5.4 kB)
Collecting cycler>=0.10 (from matplotlib->mzapy)
  Using cached cycler-0.12.1-py3-none-any.whl.metadata (3.8 kB)
Collecting fonttools>=4.22.0 (from matplotlib->mzapy)
  Downloading fonttools-4.55.0-cp312-cp312-macosx_10_13_universal2.whl.metadata (164 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 164.5/164.5 kB 4.8 MB/s eta 0:00:00
Collecting kiwisolver>=1.3.1 (from matplotlib->mzapy)
  Using cached kiwisolver-1.4.7-cp312-cp312-macosx_11_0_arm64.whl.metadata (6.3 kB)
Collecting packaging>=20.0 (from matplotlib->mzapy)
  Downloading packaging-24.2-py3-none-any.whl.metadata (3.2 kB)
Collecting pillow>=8 (from matplotlib->mzapy)
  Downloading pillow-11.0.0-cp312-cp312-macosx_11_0_arm64.whl.metadata (9.1 kB)
Collecting pyparsing>=2.3.1 (from matplotlib->mzapy)
  Downloading pyparsing-3.2.0-py3-none-any.whl.metadata (5.0 kB)
Collecting python-dateutil>=2.7 (from matplotlib->mzapy)
  Using cached python_dateutil-2.9.0.post0-py2.py3-none-any.whl.metadata (8.4 kB)
Collecting pytz>=2020.1 (from pandas->mzapy)
  Using cached pytz-2024.2-py2.py3-none-any.whl.metadata (22 kB)
Collecting tzdata>=2022.7 (from pandas->mzapy)
  Downloading tzdata-2024.2-py2.py3-none-any.whl.metadata (1.4 kB)
Collecting six>=1.5 (from python-dateutil>=2.7->matplotlib->mzapy)
  Using cached six-1.16.0-py2.py3-none-any.whl.metadata (1.8 kB)
Downloading h5py-3.12.1-cp312-cp312-macosx_11_0_arm64.whl (2.9 MB)
   ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 2.9/2.9 MB 8.3 MB/s eta 0:00:00
Downloading numpy-2.1.3-cp312-cp312-macosx_14_0_arm64.whl (5.1 MB)
   ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 5.1/5.1 MB 7.6 MB/s eta 0:00:00
Using cached hdf5plugin-5.0.0-py3-none-macosx_10_13_universal2.whl (14.0 MB)
Using cached matplotlib-3.9.2-cp312-cp312-macosx_11_0_arm64.whl (7.8 MB)
Downloading pandas-2.2.3-cp312-cp312-macosx_11_0_arm64.whl (11.4 MB)
   ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 11.4/11.4 MB 8.2 MB/s eta 0:00:00
Using cached scipy-1.14.1-cp312-cp312-macosx_14_0_arm64.whl (23.1 MB)
Downloading contourpy-1.3.1-cp312-cp312-macosx_11_0_arm64.whl (255 kB)
   ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 255.4/255.4 kB 10.0 MB/s eta 0:00:00
Using cached cycler-0.12.1-py3-none-any.whl (8.3 kB)
Downloading fonttools-4.55.0-cp312-cp312-macosx_10_13_universal2.whl (2.8 MB)
   ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 2.8/2.8 MB 10.5 MB/s eta 0:00:00
Using cached kiwisolver-1.4.7-cp312-cp312-macosx_11_0_arm64.whl (63 kB)
Downloading packaging-24.2-py3-none-any.whl (65 kB)
   ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 65.5/65.5 kB 7.0 MB/s eta 0:00:00
Downloading pillow-11.0.0-cp312-cp312-macosx_11_0_arm64.whl (3.0 MB)
   ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 3.0/3.0 MB 11.2 MB/s eta 0:00:00
Downloading pyparsing-3.2.0-py3-none-any.whl (106 kB)
   ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 106.9/106.9 kB 9.0 MB/s eta 0:00:00
Using cached python_dateutil-2.9.0.post0-py2.py3-none-any.whl (229 kB)
Using cached pytz-2024.2-py2.py3-none-any.whl (508 kB)
Downloading tzdata-2024.2-py2.py3-none-any.whl (346 kB)
   ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 346.6/346.6 kB 9.5 MB/s eta 0:00:00
Using cached six-1.16.0-py2.py3-none-any.whl (11 kB)
Building wheels for collected packages: mzapy
  Building wheel for mzapy (pyproject.toml) ... done
  Created wheel for mzapy: filename=mzapy-1.8.dev3-py3-none-any.whl size=27034 sha256=417f356cf1248c0630734298d8b3f89db675a0ae2345c6ee4c73bfe26e4f8560
  Stored in directory: /private/var/folders/wy/94m7dyf92f519_z6_pt64cg80000gn/T/pip-ephem-wheel-cache-549iz57l/wheels/d2/75/a7/05d1a9e17850cc85f1993616f9dcf7b398aca93d0ad097f159
Successfully built mzapy
Installing collected packages: pytz, tzdata, six, pyparsing, pillow, packaging, numpy, kiwisolver, fonttools, cycler, scipy, python-dateutil, h5py, contourpy, pandas, matplotlib, hdf5plugin, mzapy
Successfully installed contourpy-1.3.1 cycler-0.12.1 fonttools-4.55.0 h5py-3.12.1 hdf5plugin-5.0.0 kiwisolver-1.4.7 matplotlib-3.9.2 mzapy-1.8.dev3 numpy-2.1.3 packaging-24.2 pandas-2.2.3 pillow-11.0.0 pyparsing-3.2.0 python-dateutil-2.9.0.post0 pytz-2024.2 scipy-1.14.1 six-1.16.0 tzdata-2024.2

[notice] A new release of pip is available: 24.0 -> 24.3.1
[notice] To update, run: pip install --upgrade pip

(testenv) ╭─ross200@WE50940 ~/Desktop/mzapy_test 
╰─$ python3 -c "from mzapy import __version__; print(__version__)"                  
1.8.dev3

```

**Comment by wkumler** [comment]
Yes, that worked! I'm now able to install the new branch. Thank you for fixing that. Seems like I'm still running into issues with other things but I'll open a separate issue for those since this one has been solved.
